snippets/toff/index.js

```diff
@@ -386,7 +388,110 @@ class Oplog {
         let res = this.oplog.aggregate(p, this.options)
         print(res)
     }
-    
+
+    byTestName(testName) {
+        let lookupBeforeTestTs = {
```

Comment:
The aggregation pipeline is inspired by mongoSQL. 

The note generated by appendOplogNote looks like 
```
{
op: 'n',
ns: '',
o: {
fixture: 'ClusterToClusterReplication',
hook: 'before_test',
test: 'jstests/core/query/basic6.js'
},
ts: Timestamp({ t: 1699632506, i: 2 }),
t: Long("1"),
v: Long("2"),
wall: ISODate("2023-11-10T16:08:26.035Z")
},
```

I have to use two lookup stage to append two new fields with the `ts` value retrieved from the `before_test` doc and `after_test` doc in order to apply the match stage later.